### ISSUE TITLE
Nylas sending email multiple times 

### ISSUE THREAD

**Issue by bengotow** [nylas/N1#2695]
As reported in: https://nylas.zendesk.com/agent/tickets/5141

**Comment by whadibrata** [comment]
I have only realized this bug today after sending many duplicate emails over the last two days.
Until the bug is completely resolved, I think the simple solution is to disable link tracking and open tracking.

**Comment by yishus** [comment]
+1

it's not showing on N1 that multiple emails are sent out. But it shows up when I am on gmail web app/mobile app.

**Comment by smacking** [comment]
+1

**Comment by davidbgeek** [comment]
+1

**Comment by stolinski** [comment]
+1

**Comment by smacking** [comment]
The main question is: are people really receiving our emails in duplicate? On "Sent Mail", that's what I see, but I am not sure if that's really happening...

**Comment by grinich** [comment]
My hunch here is that the sync engine is NOT actually sending two messages— it's sending a single message but then ALSO saving a second message to the sent folder. 

This behavior is needed for many old-school IMAP mail servers which don't automatically create a message in the Sent folder when you actually send something. (Remember, traditionally IMAP and SMTP servers operate on different hosts.) To compensate for this, clients like Mail.app will have actual settings for whether to save mail on the server. 
![image](https://cloud.githubusercontent.com/assets/92417/17604671/a42fcff8-5fcb-11e6-948f-baf0eba027d4.png)

The is no actual API for determining whether an SMTP server will automatically store a copy of the sent message in the user's mailbox Sent folder. I believe we have this hard-coded in our settings for the sync engine per-provider. e.g. We _don't_ create a copy in the Sent folder for Gmail because we know it does that automatically.

Our current sending behavior does this:
1. Send
2. (Optionally) Save a copy to Sent folder

We need to change it to this:
1. Send. Keep track of the Message-Id for what we sent
2. Wait a few moments
3. Check the Sent folder
4. Only create a copy of the sent message if a message with the previous Message-Id does not exist in that folder.

This should eliminate the "double sending" issue that people are reporting. 

If we are actually sending two messages, that is another issue that we really need to resolve as well.

**Comment by jstejada** [comment]
@grinich oh interesting, this was also reported here: https://github.com/nylas/N1/issues/2709

**Comment by grinich** [comment]
@jstejada yeah — I really feel that we shouldn't make it optional for saving an item to the Sent folder. Unchecking that box accidentally just confuses people. There should always be one message saved to the Sent folder... but never two.

**Comment by jstejada** [comment]
Yeah definitely, makes sense -- merging that issue into this one and updating the sync-engine issue

**Comment by emorikawa** [comment]
This is related to https://github.com/nylas/N1/issues/1457 and https://phab.nylas.com/T6958

**Comment by bengotow** [comment]
Reported by another user today via ZenDesk. Internal link: https://nylas.zendesk.com/agent/tickets/6172
